src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/compute/v5_4/OpenNebulaComputePluginTest.java

```diff
@@ -97,761 +97,441 @@
-	public void testRequestInstanceSuccessfulWithoutNetworksId() throws FogbowException {
+	public void testUpdateHardwareRequirementsNoNewFlavors() throws UnexpectedException {
 		// set up
-		Client client = Mockito.mock(Client.class);
-		PowerMockito.mockStatic(OpenNebulaClientUtil.class);
-		BDDMockito.given(OpenNebulaClientUtil.createClient(Mockito.anyString(), Mockito.anyString()))
-				.willReturn(client);
+		TemplatePool templatePool = Mockito.mock(TemplatePool.class);
+		Template template = Mockito.mock(Template.class);
+		Iterator<Template> templateIterator = Mockito.mock(Iterator.class);
+		String flavorId = this.hardwareRequirements.getFlavorId();
 
-		LaunchCommandGenerator mockLaunchCommandGenerator = Mockito.spy(new DefaultLaunchCommandGenerator());
-		Mockito.doReturn(FAKE_USER_DATA).when(mockLaunchCommandGenerator).createLaunchCommand(Mockito.any());
-		this.plugin.setLaunchCommandGenerator(mockLaunchCommandGenerator);
-		
-		CloudUser cloudUser = createCloudUser();
-		List<String> networkIds = null;
-		ComputeOrder computeOrder = createComputeOrder(networkIds, CPU_VALUE_1, MEMORY_VALUE_2048, DISK_VALUE_8GB);
-		HardwareRequirements flavor = createHardwareRequirements();
-		Mockito.doReturn(flavor).when(this.plugin).findSmallestFlavor(client, computeOrder);
+		Mockito.doReturn(new HashMap<String, String>()).when(this.plugin).getImagesSizes(Mockito.any(Client.class));
+		Mockito.doReturn(DISK_VALUE_30GB).when(this.plugin).getDiskSizeFromImageSizeMap(Mockito.any(Map.class), Mockito.anyString());
+		Mockito.doReturn(true).when(this.plugin).containsFlavor(Mockito.any(HardwareRequirements.class));
+		PowerMockito.when(OpenNebulaClientUtil.getTemplatePool(Mockito.any(Client.class))).thenReturn(templatePool);
 
-		OneResponse response = Mockito.mock(OneResponse.class);
-		PowerMockito.mockStatic(VirtualMachine.class);
-		PowerMockito.when(VirtualMachine.allocate(Mockito.any(Client.class), Mockito.anyString())).thenReturn(response);
-		Mockito.when(response.isError()).thenReturn(false);
-		
-		int choice = 1;
-		String networkId = null;
-		String valueOfCpu = String.valueOf(CPU_VALUE_4);
-		String valueOfRam = String.valueOf(MEMORY_VALUE_2048);
-		String valueOfDisk = String.valueOf(DISK_VALUE_8GB);
-		String template = generateTemplate(choice, valueOfCpu, valueOfRam, networkId, valueOfDisk);
+		Mockito.when(templatePool.iterator()).thenReturn(templateIterator);
+		Mockito.when(templateIterator.hasNext()).thenReturn(true, false);
+		Mockito.when(templateIterator.next()).thenReturn(template);
+		Mockito.when(template.getId()).thenReturn(flavorId);
+		Mockito.when(template.getName()).thenReturn(this.hardwareRequirements.getName());
+		Mockito.when(template.xpath(TEMPLATE_CPU_PATH)).thenReturn(String.valueOf(this.hardwareRequirements.getCpu()));
+		Mockito.when(template.xpath(TEMPLATE_MEMORY_PATH)).thenReturn(String.valueOf(this.hardwareRequirements.getMemory()));
+		Mockito.when(template.xpath(TEMPLATE_DISK_SIZE_PATH)).thenReturn("0");
```

Comment:
The same previous suggestion, use a constant.

Reply:
done